fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -854,6 +854,47 @@ private void checkAssignedTargetIndexName(String baseIndexName, String targetInd
         }
     }
 
+    private boolean checkPartitionColumnChangeAllowed(Column oldPartitionColumn, Column newPartitionColumn) {
+        if (oldPartitionColumn == null || newPartitionColumn == null) {
+            return false;
+        }
+        if (!oldPartitionColumn.getName().equalsIgnoreCase(newPartitionColumn.getName())) {
+            return false;
+        }
+        if (!oldPartitionColumn.getType().equals(newPartitionColumn.getType())) {
+            return false;
+        }
+        if (oldPartitionColumn.isAllowNull() != newPartitionColumn.isAllowNull()) {
+            return false;
+        }
+
+        if (oldPartitionColumn.getStrLen() != newPartitionColumn.getStrLen()) {
+            return false;
+        }
+        if (oldPartitionColumn.getPrecision() != newPartitionColumn.getPrecision()) {
+            return false;
+        }
+        if (oldPartitionColumn.getScale() != newPartitionColumn.getScale()) {
+            return false;
+        }
+        if (oldPartitionColumn.isKey() != newPartitionColumn.isKey()
+                || (oldPartitionColumn.getAggregationType() != newPartitionColumn.getAggregationType())) {
+            // change partition column from key to value is allowed
```

Comment:
it think this comment should describe why new partition column could be value with AggregateType.NONE